src/Sulu/Bundle/AdminBundle/Resources/js/views/List/toolbarActions/UploadToolbarAction.js

```diff
@@ -0,0 +1,307 @@
+import symfonyRouting from 'fos-jsrouting/router';
+import {translate} from '../../../utils/Translator';
+import AbstractListToolbarAction from './AbstractListToolbarAction';
+
+export default class UploadToolbarAction extends AbstractListToolbarAction {
+    @observable dropzoneRef: ?DropzoneRef;
+    @observable errors: string[] = [];
+
+    @action setDropzoneRef = (ref: ?DropzoneRef) => {
+        this.dropzoneRef = ref;
+    };
+
+    @action handleClick = () => {
+        const {dropzoneRef} = this;
+
+        if (dropzoneRef) {
+            dropzoneRef.open();
+            this.removeErrors();
+        }
+    };
+
+    removeErrors = () => {
+        for (const error of this.errors) {
+            this.removeError(error);
+        }
+    };
+
+    @action removeError = (errorToRemove: string) => {
+        this.errors = this.errors.filter((existingError) => errorToRemove !== existingError);
+        this.list.errors = this.list.errors.filter((existingError) => errorToRemove !== existingError);
+    };
+
+    @action addError = (error: string) => {
+        this.removeError(error);
+
+        this.errors = [...this.errors, error];
+        this.list.errors = [...this.list.errors, error];
+    };
+
+    handleError = (fileRejections: FileRejection[]) => {
+        for (const fileRejection of fileRejections) {
+            for (const {code} of fileRejection.errors) {
+                let error;
+                switch (code) {
+                    case 'file-invalid-type':
+                        error = translate('sulu_admin.dropzone_error_file-invalid-type', {
+                            fileName: fileRejection.file.name,
+                            allowedTypes: this.accept ? this.accept.join(', ') : undefined,
+                        });
+                        break;
+                    case 'file-too-large':
+                        error = translate('sulu_admin.dropzone_error_file-too-large', {
+                            fileName: fileRejection.file.name,
+                            maxSize: this.maxSize ? this.getReadableFileSizeString(this.maxSize) : undefined,
+                        });
+                        break;
+                    case 'file-too-small':
+                        error = translate('sulu_admin.dropzone_error_file-too-small', {
+                            fileName: fileRejection.file.name,
+                            minSize: this.minSize ? this.getReadableFileSizeString(this.minSize) : undefined,
+                        });
+                        break;
+                    case 'too-many-files':
+                        error = translate('sulu_admin.dropzone_error_too-many-files', {
+                            fileName: fileRejection.file.name,
+                            maxFiles: this.maxFiles,
+                        });
+                        break;
+                    default:
+                        error = translate('sulu_admin.an_error_occurred');
```

Comment:
would use a more specific `sulu_admin.unexpected_upload_error` that says something like `Unexpected error while uploading the file ... to the server`